spmd/tensor/__init__.py

```diff
@@ -1,7 +1,82 @@
 # Copyright (c) Meta Platforms, Inc. and affiliates
+from typing import Optional, Sequence, cast
+
+import torch
 from spmd.tensor.api import DTensor
-from spmd.tensor.device_mesh import DeviceMesh
+from spmd.tensor.device_mesh import DeviceMesh, get_global_device_mesh
 from spmd.tensor.placement_types import Placement, Shard, Replicate, _Partial
 
+
 # Import all builtin dist tensor ops
 import spmd.tensor.ops
+
+
+def distribute_tensor(
+    tensor: torch.Tensor,
+    device_mesh: Optional[DeviceMesh] = None,
+    placements: Optional[Sequence[Placement]] = None,
+) -> DTensor:
+    """
+    Distribute a torch.Tensor to the `device_mesh` according to the `placements`
+    specified. The rank of `device_mesh` and `placements` must be the same.
+
+    Args:
+        tensor (torch.Tensor): torch.Tensor to be distributed
+        device_mesh (:class:`DeviceMesh`, optional): DeviceMesh to distribute the
+            tensor, if not specified, must be called under a DeviceMesh context
+            manager, default: None
+        placements (List[:class:`Placement`], optional): the placements that
+            describes how to place the tensor on DeviceMesh, must have the same
+            number of elements as `device_mesh.ndim`. If not specified, we will
+            by default replicate the tensor across the `device_mesh` from the
+            first rank of each dimension of the `device_mesh`.
+
+    Returns:
+        A :class:`DTensor` object
+    """
+    # get default device mesh if there's nothing specified
+    device_mesh = (
+        get_global_device_mesh() if device_mesh is None else device_mesh
+    )
+    # convert tensor to the correponding device type if it's not in that device type
+    tensor = tensor.to(device_mesh.device_type)
+    # set default placements to replicated if not specified
+    if placements is None:
+        placements = [Replicate() for _ in range(device_mesh.ndim)]
+
+    # distribute the tensor according to PlacementSpec
```

Comment:
If each local tensor contributes to form a DTensor, what is the use case for calling scatter and broadcast?

Reply:
The only way we allow user to contribute a local tensor to DT is `from_local`, we don't allow `DTensor` constructor directly. If there're existing parameters, we need to shard them and do scatter/broadcast